pkg/networkservice/core/eventchannel/monitor_connection_server.go

```diff
@@ -39,23 +37,15 @@ func NewMonitorConnectionMonitorConnectionsServer(ctx context.Context, eventCh c
 		ctx:     ctx,
 		eventCh: eventCh,
 	}
-	go func() {
-		<-ctx.Done()
-		// Funnel through Send here to trigger close in *one* place to avoid race conditions
-		_ = rv.Send(&networkservice.ConnectionEvent{})
-	}()
 	return rv
 }
 
 func (m *monitorConnectionMonitorConnectionsServer) Send(event *networkservice.ConnectionEvent) error {
-	select {
```

Comment:
The root cause is this `select`.  Actually two goroutines can read and close `eventCh` in one moment.

Reply:
Elegant solution :)